### ISSUE TITLE
Service Bus Function processing same message over and over (hundreds of times)

### ISSUE THREAD

**Issue by cgraus** [Azure/azure-functions-host#6594]
I have an Azure function which I just updated to look like this:

  public void Run([ServiceBusTrigger("xxxx", Connection = "ServiceBusConnectionString", IsSessionsEnabled = true)] Message message, ILogger log, IMessageSession messageSession)

I did this so I could do this:

  var rec = messageSession as MessageReceiver;

            if(rec != null)
            {
                rec.RetryPolicy = Microsoft.ServiceBus.RetryPolicy.NoRetry;
            }

            _event.PushEvent(new ATCEvent
            {
                EventName = $"{message.MessageId} - {message.SystemProperties.DeliveryCount}",
                EventData = string.Empty,
                DateOfEvent = DateTime.Now, 
                EventLevel = 1,
                System = Systems.CRMSQL
            }).Wait();
            
            // Manually close message so it won't requeue
            messageSession.CloseAsync().Wait();

My event logging is showing the same message being processed over and over

EventName	EventData	EventLevel	DateOfEvent
5f353e313f104c298bdffa08030d9daa - 1		1	2020-09-03 01:17:47.903
EventName	EventData	EventLevel	DateOfEvent
5f353e313f104c298bdffa08030d9daa - 1		1	2020-09-03 01:17:46.083
EventName	EventData	EventLevel	DateOfEvent
5f353e313f104c298bdffa08030d9daa - 1		1	2020-09-03 01:17:45.227

Why is my function pulling the same message over and over, without the delivery count increasing, even though I am manually closing the message? It did this before I added this code, this is the issue I was trying to resolve.  

My logs confirm we called the function to push into the service bus once.  I log pushing as an event, there is one event log.

![image](https://user-images.githubusercontent.com/5778834/92061063-b397a480-edd8-11ea-9a37-7a9014147760.png)

This is the graph of messages into that queue.

**Comment by ThejaChoudary** [comment]
@cgraus can you please provide us the repro steps and name of function app, time stamp & region?

**Comment by cgraus** [comment]
The issue was that my process was taking longer than 5 minutes, it was about 500 inserts into CRM.  So I changed the code to cancel the message so it doesn't get resent.  Finding accurate documentation is always difficult because things change all the time, but finally I found this injection signature:

Message message, string lockToken, IMessageSession messageSession)
        {
            // Manually complete message so it won't requeue
            await messageSession.CompleteAsync(lockToken);

I couldn't find a way to inject a MessageReceiver anymore, meaning I can't set the retry policy, but I believe this is working fine now, thank you.
